src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -39,7 +39,7 @@
 
     public QueryServiceConnection(String url, Properties properties) {
         this.properties = properties;
-        this.serviceRootUrl = getServiceRootUrl(url);
+        this.serviceRootUrl = url.substring(Constants.DATASOURCE_TYPE.length());
```

Comment:
this will not work if the login url ends with /.

Reply:
fixed